File: utils/data_manager.py
import json
import os

DATA_FILE = 'data.json'

DEFAULT_DATA = {
    "guilds": {}
}

def load_data():
    if not os.path.exists(DATA_FILE):
        save_data(DEFAULT_DATA)
        return DEFAULT_DATA
    try:
        with open(DATA_FILE, 'r') as f:
            return json.load(f)
    except (json.JSONDecodeError, IOError):
        return DEFAULT_DATA

def save_data(data):
    with open(DATA_FILE, 'w') as f:
        json.dump(data, f, indent=4)

def get_guild_data(guild_id):
    data = load_data()
    guild_id = str(guild_id)
    if guild_id not in data["guilds"]:
        data["guilds"][guild_id] = {
            "prefix": "!",
            "security": {
                "anti_spam": False,
                "anti_link": False
            },
            "mod_logs": [],
            "voice_settings": {
                "category_id": None,
                "interface_channel_id": None
            },
            "greet_channel": None,
            "greet_message": "Welcome {member.mention} to the server!"
        }
        save_data(data)
    return data["guilds"][guild_id]

def update_guild_data(guild_id, key, value):
    data = load_data()
    guild_id = str(guild_id)
    if guild_id not in data["guilds"]:
        get_guild_data(guild_id)
        data = load_data()

    data["guilds"][guild_id][key] = value
    save_data(data)

File: src/utils/dataManager.js
const fs = require('fs');
const path = require('path');

const DATA_FILE = path.join(__dirname, '../../data.json');

const defaultData = {
    guilds: {}
};

function loadData() {
    if (!fs.existsSync(DATA_FILE)) {
        saveData(defaultData);
        return defaultData;
    }
    try {
        const raw = fs.readFileSync(DATA_FILE);
        return JSON.parse(raw);
    } catch (e) {
        return defaultData;
    }
}

function saveData(data) {
    fs.writeFileSync(DATA_FILE, JSON.stringify(data, null, 4));
}

function getGuildData(guildId) {
    const data = loadData();
    if (!data.guilds[guildId]) {
        data.guilds[guildId] = {
            security: {
                antiSpam: false,
                antiLink: false
            },
            greetChannel: null,
            greetMessage: "Welcome to the server, {member}!",
            voiceSettings: {
                categoryId: null,
                interfaceChannelId: null
            },
            modLogsChannel: null
        };
        saveData(data);
    }
    return data.guilds[guildId];
}

function updateGuildData(guildId, key, value) {
    const data = loadData();
    if (!data.guilds[guildId]) {
        getGuildData(guildId);
    }
    data.guilds[guildId][key] = value;
    saveData(data);
}

module.exports = {
    getGuildData,
    updateGuildData
};
